### PR TITLE
Specter2: Eng resources

### PR DESCRIPTION
**Pull request by lynnerang**
Imports some new text components with links (since we can't build links via the API) and spits them into the legend header.  

Note: Mercado MVP is the only one with the mobile vs web distinction (as far as I know), but figured I'd use the same format for all to keep things more simple and scalable.

### CODE REVIEW

File: specter2/src/constants.ts [hunk 1]
```diff
@@ -779,23 +779,45 @@ const CUSTOM_OPTS = [
 ];
 
 /**
- * @description An array of component keys for importing instruction panels.
+ * @description Component keys for importing instruction panels.
  *
  * @kind constant
  * @name INSTRUCTION_PANEL_KEYS
- * @type {Array}
+ * @type {ArrObjectay}
```
Comment (gblakeman) [inline specter2/src/constants.ts:786]: Should this be `@type {Object}`?

**Review by gblakeman — APPROVED** [review]
🛳💥

File: specter2/src/Painter.ts [hunk 2]
```diff
@@ -724,6 +726,17 @@ export default class Painter {
...
       const legend = slide.findOne(({ name }) => name.includes('Spec Legend')) as FrameNode;
+      const dsLibrary: Library = getDsLibrary(libraries);
+
+      // add eng resource links for the DS library used
+      if (dsLibrary) {
+        const legendInstructions = legend.findOne(({ name }) => name.includes('Instructions')) as FrameNode;
+        const importPlatform: string = platform === 'web' ? 'web' : 'mobile';
+        const importKey: string = MISC_IMPORT_KEYS[`${dsLibrary.replace('-', '')}Resources`][importPlatform];
+        const libraryResources: InstanceNode = await importInstance(importKey);
+        legendInstructions.appendChild(libraryResources);
```
Comment (joeyyang) [inline specter2/src/Painter.ts:737]: where does `legendInstructions` get consumed? Not seeing it referenced anywhere, maybe I'm missing something.

**Review by joeyyang — APPROVED** [review]
A few nits otherwise looks good!

File: specter2/src/Painter.ts (hunk 2 above)
Comment (lynnerang) [reply specter2/src/Painter.ts:737]: not consumed anywhere, but we have to find that frame within the imported legend to append eng resources to (on line 737)

File: specter2/src/constants.ts [hunk 3]
```diff
@@ -779,23 +779,45 @@ const CUSTOM_OPTS = [
...
- * @type {Array}
+ * @type {Object}
  */
 const INSTRUCTION_PANEL_KEYS = {
   notesPanel: 'e2abd4594dd4c3bfc73887052d30d3918fcf288e',
   engInstructionPanel: 'd0cd1f1c478cafb79d51096b4503d3cf0f1f2be4',
 };
 
+/**
+ * Component keys for importing miscellaneous content.
```
[the remark this answers is not in the document]
Comment (lynnerang) [reply specter2/src/constants.ts:794]: every component in Figma has a 'key' property that's required for the import, so these are the component keys we need to store to import instances of these components.  I can add that in the desc

**Pull request merged**